### ISSUE TITLE
[IMPROVEMENT] Add onsale-products listing page

### ISSUE THREAD

**Issue by Hlavtox** [PrestaShop/PrestaShop#15384]
**Scenario**
I think that most of merchants use "ON SALE" flag:
- to get rid of products, that they will no longer sell
- the supplier stopped manufacturing it
- to get rid of slightly damaged item or an item with a missing packaging/manual etc.
- items that have been returned in no-reason period (by law, Czech Republic) and thus cannot be sold normally

People are used to this and when they want to get quality product, usually undamaged, but at lower price, they look for "ON SALE" section of the web page.

**Solution**
Make a page, that lists all products, that have onsale flag turned on.

**Technically**
Basically it's just a copy of newproducts controller. Other than naming, only code change is in Product.php, you are not getting products with `date < 30 days`, but products with `onsale = 1`.
```
add controllers/front/listing/OnsaleProductsController.php
add src/Adapter/OnsaleProducts/OnsaleProductsProductSearchProvider.php
add getOnsaleProducts function to classes/Product.php
add a template
add a redirect rule
add a wording
add it to class autoload
```

**Comment by khouloudbelguith** [comment]
Hi @Hlavtox,

Thanks for your report.
Ping @marionf @colinegin what do you think about hs improvement?
We can add the "On sale flag" on the Product details page => Pricing Tab"
![image](https://user-images.githubusercontent.com/16067358/64178243-04d52700-ce59-11e9-8e9f-2ac4d0fceb67.png)

Thanks!

**Comment by marionf** [comment]
Hello @Hlavtox 

There is already a "price-drop" page with all discounted products 
I think usually, "on sale" flag is used for discounted products
By the way, it's not legal to display an "on sale" flag if the product isn't discounted

**Comment by Hlavtox** [comment]
@marionf 

Price-drop displays only discounted products. As you say, on-sale is something completely different. You usually put on-sale to stuff you want to get rid of and sell as a priority.

What's legal or not depends on the state you sell in and also, it depend's on the wording. On sale = selling out of stock, it doesn't neccessarily mean the stuff is discounted, you can just indicate that you have last pieces in stock and that it won't be available any longer.

PS: If "SALE" was not legal, they would need to close 90% of stores in Czech republic... :-D You come to an electronic store and see TV with SUPER EXTRA 50% DISCOUNT, you check online and it's sold for the same price in 50 other stores... :-)

**Comment by MatShir** [comment]
Hi @Daniel Hlavacek your request is interesting but the example you gave us are pages with products with reduced price (there are either discounts or on SALE). That looks like a lot as the drop price page.

for these cases:

> I think that most of merchants use "ON SALE" (Outlet) flag:
> - to get rid of products, that they will no longer sell
> - the supplier stopped manufacturing it
> - products that are from last-season collection
> - to get rid of slightly damaged item or an item with a missing packaging/manual etc.
> - items that have been returned and thus cannot be sold normally

Would it be more interesting to create specific categories and linked to the menu?

**Comment by Hlavtox** [comment]
@MatShir Hi, it's very simmilar, but still different.

If the product is discounted, it does not mean it's on sale. That's what the on_sale flag is for - “we need to get rid of it, last chance”.  

In some stores, everything is under some discount most of the times, and it does not mean it's the last chance to buy it.

**Comment by Junebyun** [comment]
Hi there, @Hlavtox thank you for your suggestions! But I am not sure if creating another generic page for the products you have mentioned is necessary for all merchants, for me it seems to be a specific need. 

I think two options could be considered for your scenario:

- create a subcategory that you could call "_Outlet_" or "_Last chance_" in the "_Prices drop_" page (soon going to be named as "_Sale_"), to gather the mentioned products that have a discount.

- or create a category called "_Outlet_" or "_Last chance_", which would appear in the menu, for the mentioned products. You could separate products in subcategories if you wish to differentiate those which have discounts with those which don't).

From the research I did, products that have the "_on sale_" tag are more understood as products which have reduced prices, instead of this notion of "_products that have to be get rid of or last chance_". 

If you find this terminology confusing, you can change the label of this tag in IMPROVE > International > Translations > choose Themes translations in the drop-down list > type "On sale!".

**Comment by Hlavtox** [comment]
@Junebyun Creating a category for Outlet items is kind of a hackjob. 

- More work. You would need to remember to add it to this special category AND tick the on_sale flag.
- Hard to apply custom styling to this category without modifying the template.

I have done it in my Prestashop and I think it would be useful, take it or leave it. 😄
